pubsub/src/subscription.rs

```diff
@@ -290,19 +294,25 @@ where
 				let unsub = self.unsubscribe.clone();
 				let notification = self.notification.clone();
 				let subscribe_future = rx.map_err(|_| subscription_rejected()).and_then(move |result| {
-					futures::done(match result {
+					futures::future::ready(match result {
 						Ok(id) => {
 							session.add_subscription(&notification, &id, move |id| {
-								let _ = unsub.call(id, None).wait();
+								// TODO [ToDr] We currently run unsubscribe tasks on a shared thread pool.
```

Comment:
Might be worth adding a follow up issue for this otherwise this TODO will be here forever

Reply:
Done, see #570